src/app/body/all-products/all-products/all-products.component.ts

```diff
@@ -1,19 +1,23 @@
 import { Component, OnInit } from '@angular/core';
-import { AngularFireDatabase } from '@angular/fire/database';
-import { Observable } from 'rxjs';
+import { ProductService } from '../../../services/product.service';
+import { Product } from '../../../models/product.model';
 
 @Component({
   selector: 'app-all-products',
   templateUrl: './all-products.component.html',
   styleUrls: ['./all-products.component.scss']
 })
 export class AllProductsComponent implements OnInit {
-  public products: Observable<any[]>;
-  constructor(public db: AngularFireDatabase) {
-    this.products = db.list('products').valueChanges();
-  }
-
+  public products: Product[];
+  constructor(private productService: ProductService) { }
   ngOnInit(): void {
+    this.getProducts();
```

Comment:
You don't have to get products one more time. You should take it from snapshot.

Reply:
fixed